lib/src/test/java/vmath/Mat4Test.java

```diff
@@ -194,4 +194,123 @@ void givenInvalidIndices_whenGetIsCalled_thenThrowsException() {
         assertThrows(IndexOutOfBoundsException.class, () -> m.get(0, -1));
         assertThrows(IndexOutOfBoundsException.class, () -> m.get(0, 4));
     }
-}
+
+    @Test
+    void givenIdentityMatrix_whenMultipliedWithAnyMatrix_thenResultIsTheSame() {
+        Mat4 identity = new Mat4(); 
+        Mat4 someMatrix = new Mat4(
+                1.0f, 2.0f, 3.0f, 4.0f,
+                5.0f, 6.0f, 7.0f, 8.0f,
+                9.0f, 10.0f, 11.0f, 12.0f,
+                13.0f, 14.0f, 15.0f, 16.0f
+        );
+
+        Mat4 result = identity.mul(someMatrix);
+        
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                assertEquals(result.get(i, j), someMatrix.get(i, j));
+            }
+        }
+    }
+
+    @Test
+    void givenTwoSpecificMatrices_whenMultiplied_thenResultIsAsExpected() {
+        Mat4 matrixA = new Mat4(
+                1.0f, 2.0f, 3.0f, 4.0f,
+                5.0f, 6.0f, 7.0f, 8.0f,
+                9.0f, 10.0f, 11.0f, 12.0f,
+                13.0f, 14.0f, 15.0f, 16.0f
+        );
+
+        Mat4 matrixB = new Mat4(
+                2.0f, 0.0f, 0.0f, 0.0f,
+                0.0f, 2.0f, 0.0f, 0.0f,
+                0.0f, 0.0f, 2.0f, 0.0f,
+                0.0f, 0.0f, 0.0f, 2.0f
+        );
+
+        Mat4 expected = new Mat4(
+                2.0f, 4.0f, 6.0f, 8.0f,
+                10.0f, 12.0f, 14.0f, 16.0f,
+                18.0f, 20.0f, 22.0f, 24.0f,
+                26.0f, 28.0f, 30.0f, 32.0f
+        );
+
+        Mat4 result = matrixA.mul(matrixB);
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                assertEquals(expected.get(i, j), result.get(i, j));
+            }
+        }
+    }
+
+    @Test
+    void givenDifferentSizedMatrices_whenMultiplied_thenThrowsException() {
+        Mat4 matrixA = new Mat4();
+        Mat4 matrixB = new Mat4(
+                1.0f, 2.0f, 3.0f, 4.0f, 
+                5.0f, 6.0f, 7.0f, 8.0f,
+                9.0f, 10.0f, 11.0f, 12.0f,
+                13.0f, 14.0f, 15.0f, 16.0f
+        );
+
+        assertThrows(IllegalArgumentException.class, () -> matrixA.mul(matrixB));
+    }
```

Comment:
Two tests are failing in the gradle build: this and 'givenMatricesFromFloatArray_whenDifferentSizedMatricesMultiplied_thenThrowException()'. Looking at the code, it appears that "mul" does not throw an exception, meaning the "assert" in these tests will be false. Please consider what you are testing here and whether it is necessary, when the constructor already checks for invalid sizes. I think it is not.